operator/builtin/parser/trace/trace_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/testutil"
+)
+
+func TestDefaultParser(t *testing.T) {
+	traceParserConfig := NewTraceParserConfig("")
+	_, err := traceParserConfig.Build(testutil.NewBuildContext(t))
+	require.NoError(t, err)
+}
+
+func TestTraceParserParse(t *testing.T) {
+	cases := []struct {
+		name           string
+		inputRecord    map[string]interface{}
+		expectedRecord map[string]interface{}
+		expectErr      bool
+		traceId        string
+		spanId         string
+		traceFlags     string
```

Comment:
Are we overspecifying here? It seems like `inputRecord` and `expectedRecord` would be enough to test that we're getting the expected behavior.

Reply:
Euh... well you want to see that the decoding move the traces to the correct fields and parse them into bytes.